ballerina-tests/http-interceptor-tests/tests/interceptors_basic_tests.bal

```diff
@@ -502,28 +502,27 @@ function testJwtInformationDecodeErrorInRequestContext() returns error? {
         });
     string response = check jwtClient->get("/requestInterceptorJwtInformation", {"authorization": "Bearer abcd"});
     test:assertEquals(response, "Hello client");
-    if reqCtxJwtDecodeError is http:ListenerAuthError {
-        test:assertEquals((<http:ListenerAuthError>reqCtxJwtDecodeError).message(), "an error occured while decoding jwt string.");
-        error? cause = (<http:ListenerAuthError>reqCtxJwtDecodeError).cause();
-        if cause is error {
-            test:assertEquals(cause.message(), "Invalid JWT.");
-        } else {
-            test:assertFail("Expected a ListenerAuthError");
-        }
+    if reqCtxJwtDecodeError is http:ListenerError {
+        test:assertEquals((<http:ListenerError>reqCtxJwtDecodeError).message(), "no member found for key: JWT_INFORMATION");
     } else {
-        test:assertFail("Expected a ListenerAuthError");
+        test:assertFail("Expected a ListenerError");
     }
 }
 
 @test:Config{
     dependsOn: [testJwtInformationDecodeErrorInRequestContext]
 }
 function testNilJwtInformationValueInRequestContext() returns error? {
+    reqCtxJwtDecodeError = ();
     http:Client jwtClient = check new("https://localhost:" + jwtInformationInReqCtxtTestPort.toString(),
         secureSocket = {
             cert: common:CERT_FILE
         });
     string response = check jwtClient->get("/requestInterceptorJwtInformation");
     test:assertEquals(response, "Hello client");
-    test:assertTrue(reqCtxJwtDecodeError is ());
+    if reqCtxJwtDecodeError is http:ListenerError {
+        test:assertEquals((<http:ListenerError>reqCtxJwtDecodeError).message(), "no member found for key: JWT_INFORMATION");
```

Comment:
do we need casting here ?

Reply:
Yes. Module level defined variables do net get narrowed